entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -28,16 +31,23 @@ import com.egm.stellio.shared.model.NgsiLdRelationship
 import com.egm.stellio.shared.model.NgsiLdRelationshipInstance
 import com.egm.stellio.shared.model.Observation
 import com.egm.stellio.shared.model.ResourceNotFoundException
+import com.egm.stellio.shared.model.getDatasetId
+import com.egm.stellio.shared.model.isAttributeOfType
+import com.egm.stellio.shared.util.JsonLdUtils
 import com.egm.stellio.shared.util.JsonLdUtils.EGM_OBSERVED_BY
 import com.egm.stellio.shared.util.JsonLdUtils.EGM_VENDOR_ID
 import com.egm.stellio.shared.util.JsonLdUtils.JSONLD_ID
 import com.egm.stellio.shared.util.JsonLdUtils.JSONLD_TYPE
+import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_PROPERTY_TYPE
 import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_RELATIONSHIP_HAS_OBJECT
 import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_RELATIONSHIP_TYPE
 import com.egm.stellio.shared.util.JsonLdUtils.compactAndStringifyFragment
 import com.egm.stellio.shared.util.JsonLdUtils.expandJsonLdKey
 import com.egm.stellio.shared.util.JsonLdUtils.expandRelationshipType
-import com.egm.stellio.shared.util.JsonLdUtils.parseJsonLdFragment
+import com.egm.stellio.shared.util.JsonLdUtils.extractRelationshipObjectOrThrowError
+import com.egm.stellio.shared.util.JsonLdUtils.getPropertyValueFromMap
+import com.egm.stellio.shared.util.JsonLdUtils.getPropertyValueFromMapAsDateTime
+import com.egm.stellio.shared.util.JsonLdUtils.getPropertyValueFromMapAsString
```

Comment:
I would prefer not to see this low level functions coming back into `EntityService`

Reply:
I added PropertyUpdate and RelationshipUpdate models to handle this.